### PR TITLE
Task 5

### PR DESCRIPTION
**Pull request by nidhi-tagline**

### CODE REVIEW

File: task4_1.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# a recursive function which returns the nth Fibonacci number
+ans = [0 for i in range(1000)]
```
Comment (himanshu-suthar-tagline) [inline task4_1.py]: Please use dictionary for caching instead of list.

File: task5.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# find out the pairs that have sum = n
+
+numbers = [9, 4, 8, 10, 2, 4, 8, 3, 14, 4, 8]
+n = int(input('Enter number :'))
+
+uniqueNumber = set()
+for i in numbers:
+    uniqueNumber.add(i)
+uniqueNumber = list(uniqueNumber)
+combinationList = [[a,b] for a in uniqueNumber for b in uniqueNumber if a+b == n and a!=b]
```
Comment (himanshu-suthar-tagline) [inline task5.py]: Use itertools.combinations for this.

File: task4_1.py (hunk 1 above)
Comment (nidhi-tagline) [reply task4_1.py]: It is done.

File: task5.py (hunk 2 above)
Comment (nidhi-tagline) [reply task5.py]: It is done.

**Review by nidhi-tagline — COMMENTED** [review]
It is done.

**Pull request merged**